packages/eslint-plugin/tests/rules/prefer-optional-chain.test.ts

```diff
@@ -501,5 +506,94 @@ foo?.bar(/* comment */a,
         },
       },
     },
+    {
+      code: 'const foo = (bar || {}).baz;',
+      errors: [
+        {
+          messageId: 'optionalChainSuggest',
+          column: 13,
+          endColumn: 28,
+          suggestions: [
+            {
+              messageId: 'optionalChainSuggest',
+              output: 'const foo = bar?.baz;',
+            },
+          ],
+        },
+      ],
+    },
+    {
+      code: '(foo.bar || {})[baz];',
+      errors: [
+        {
+          messageId: 'optionalChainSuggest',
+          column: 1,
+          endColumn: 21,
+          suggestions: [
+            {
+              messageId: 'optionalChainSuggest',
+              output: 'foo.bar?.[baz];',
+            },
+          ],
+        },
+      ],
+    },
+    {
+      // Currently it does not suggest a fix for nested optional with empty object
+      // It shows 2 suggestions, one for the outer object and one for the inner object
```

Comment:
Nit: I think this is implied by looking at the test case? IME these comments can fall out of sync pretty quickly.

```suggestion
```

Reply:
It was not obvious to me as I was not familiar with how the fixer works when there are 2 fixes, does it only picks the first one?
I removed the comment for now.

Reply:
I forget what the default behavior in tests is but it'd be the same across all rule testers. We never customize it.